src/subclasses/ContractInteractions.ts

```diff
@@ -476,25 +468,63 @@ class ContractInteractions {
     await approveTxn.wait();
 
     const offsetTxn: ContractTransaction =
-      await offsetHelper.autoOffsetUsingToken(
+      await offsetHelper.autoOffsetExactOutToken(
         swapToken.address,
         poolAddress,
-        amount,
-        { gasLimit: GAS_LIMIT }
```

Comment:
You didn't remove them consistently across all methods, however on a second thought, I think there might be benefit to keep it so that we don't run into reversed txs. So sorry for withdrawing my first review point 🙏